LayerClient/LayerClient.py

```diff
@@ -0,0 +1,509 @@
+        """
+        Fetch an existing conversation by UUID
+
+        Parameter `conversation_uuid`: The UUID of the conversation to fetch
+
+        Return: A `Conversation` instance
+        """
+        return Conversation.from_dict(
+            self._raw_request(
+                METHOD_GET,
+                self._get_layer_uri(
+                    LAYER_URI_CONVERSATIONS,
+                    conversation_uuid,
+                ),
+            )
+        )
+
+    def delete_conversation(self, conversation_uuid):
+        """
+        Delete a conversation. Affects all users in the conversation across all
+        of their devices.
+
+        Parameter `conversation_uuid`: The uuid of the conversation to delete
+        """
+        self._raw_request(
+            METHOD_DELETE,
+            self._get_layer_uri(
+                LAYER_URI_CONVERSATIONS,
+                conversation_uuid,
+            ),
+        )
+
+    def create_conversation(self, participants, distinct=True, metadata=None):
+        """
+        Create a new converstaion.
+
+        Parameters:
+        - `participants`: An array of participant IDs (strings)
+        - `distinct`: Whether or not we should create a new conversation for
+            the participants, or re-use one if one exists. Will return an
+            existing conversation if distinct=True.
+        - `metadata`: Unstructured data to be passed through to the client.
+            This data must be json-serializable.
+
+        Return: A new `Conversation` instance
+        """
+        return Conversation.from_dict(
+            self._raw_request(
+                METHOD_POST,
+                self._get_layer_uri(
+                    LAYER_URI_CONVERSATIONS,
+                ),
+                {
+                    'participants': participants,
+                    'distinct': distinct,
+                    'metadata': metadata,
+                }
+            )
+        )
+
+    def send_message(self, conversation, sender, message_parts,
+                     notification=None):
+        """
+        Send a message to a conversation.
+
+        Parameters:
+        - `conversation`: A `LayerClient.Conversation` instance for the
+        conversation we wish to send to
+        - `sender`: A `LayerClient.Sender` instance
+        - `message_parts`: An array of `LayerClient.MessagePart` objects
+        - `notification`: Optional `PushNotification` instance.
+        """
+        request_data = {
+            'sender': sender.as_dict(),
+            'parts': [
+                part.as_dict() for part in message_parts
+            ],
+        }
+        if notification:
+            request_data['notification'] = notification.as_dict()
+
+        return Message.from_dict(
+            self._raw_request(
+                METHOD_POST,
+                self._get_layer_uri(
+                    LAYER_URI_CONVERSATIONS,
+                    conversation.uuid(),
```

Comment:
You might want to get defensive in this method and check that conversation is not none, or this could throw an exception.